server/SwipeDecision.py

```diff
@@ -84,7 +84,29 @@ def showProfile(shownUserId):
                 gender = row[4]
                 workPlace = row[5]
                 profile = row[6]
-                
+            
+            #uses set intersection to compare matches of each user to get a mutual friends list
+            currentUserMatchSet = getMatchIds(currentUserId, cursor, connection)
```

Comment:
Instead of calling the database and then twice and then comparing and then checking you can use something similar to the following query to get the mutual friends first names(test beforehand to make sure it's correct):

```Select p.firstname
	from PotentialMatch pm, Profile p
	where pm.currentUser = {shownUserId}
	and pm.matchDecision  = 1 
    and exists (
		select potentialId 
			from PotentialMatch p1
            where p1.currentUser = {currentUserId} and matchDecision = 1)
	and pm.shownUser = p.userId;
```
that way you can just run the database query once and just have one return statement